src/vsr/grid.zig

```diff
@@ -981,11 +1028,16 @@ pub fn GridType(comptime Storage: type) type {
         }
 
         fn verify_read(grid: *Grid, address: u64, cached_block: BlockPtrConst) void {
-            if (Storage != @import("../testing/storage.zig").Storage)
-                // Too complicated to do async verification
-                return;
+            assert(constants.verify);
+
+            const TestStorage = @import("../testing/storage.zig").Storage;
+            if (Storage != TestStorage) return;
 
             const actual_block = grid.superblock.storage.grid_block(address);
+            const actual_header = schema.header_from_block(actual_block);
+            const cached_header = schema.header_from_block(cached_block);
+            assert(cached_header.checksum == actual_header.checksum);
```

Comment:
Checking the checksum before the entire block buffer makes errors a little easier to figure out. (i.e. "wrong block" vs "padding mismatch")